packages/twenty-front/src/modules/ui/layout/modal/components/EnhancedModalLayout.tsx

```diff
@@ -0,0 +1,252 @@
+      case 'small':
+        return theme.spacing(2);
+      case 'medium':
+        return theme.spacing(4);
+      case 'large':
+        return theme.spacing(6);
+      default:
+        return 'auto';
+    }
+  }};
+  height: ${({ isMobile, theme }) =>
+    isMobile ? theme.modal.size.fullscreen : 'auto'};
+  max-height: ${({ isMobile }) => (isMobile ? 'none' : '90vh')};
+`;
+
+const StyledHeader = styled.div`
+  align-items: center;
+  display: flex;
+  flex-direction: row;
+  height: 60px;
+  overflow: hidden;
+  padding: ${({ theme }) => theme.spacing(5)};
+`;
+
+const StyledContent = styled.div`
+  display: flex;
+  flex: 1;
+  flex: 1 1 0%;
+  flex-direction: column;
+  overflow-y: auto;
+  padding: ${({ theme }) => theme.spacing(10)};
+`;
+
+const StyledFooter = styled.div`
+  align-items: center;
+  display: flex;
+  flex-direction: row;
+  height: 60px;
+  overflow: hidden;
+  padding: ${({ theme }) => theme.spacing(5)};
+`;
+
+const StyledBackDrop = styled(motion.div)<{
+  modalVariant: ModalVariants;
+}>`
+  align-items: center;
+  background: ${({ theme, modalVariant }) =>
+    modalVariant === 'primary'
+      ? theme.background.overlay
+      : theme.background.overlay2};
+  display: flex;
+  height: 100%;
+  justify-content: center;
+  left: 0;
+  position: fixed;
+  top: 0;
+  width: 100%;
+  z-index: 9999;
+  user-select: none;
+`;
+
+/**
+ * Modal components
+ */
+type ModalLayoutHeaderProps = React.PropsWithChildren & {
+  className?: string;
+};
+
+const ModalLayoutHeader = ({ children, className }: ModalLayoutHeaderProps) => (
+  <StyledHeader className={className}>{children}</StyledHeader>
+);
+
+type ModalLayoutContentProps = React.PropsWithChildren & {
+  className?: string;
+};
+
+const ModalLayoutContent = ({
+  children,
+  className,
+}: ModalLayoutContentProps) => (
+  <StyledContent className={className}>{children}</StyledContent>
+);
+
+type ModalLayoutFooterProps = React.PropsWithChildren & {
+  className?: string;
+};
+
+const ModalLayoutFooter = ({ children, className }: ModalLayoutFooterProps) => (
+  <StyledFooter className={className}>{children}</StyledFooter>
+);
+
+/**
+ * Modal
+ */
+export type ModalSize = 'small' | 'medium' | 'large';
+export type ModalPadding = 'none' | 'small' | 'medium' | 'large';
+export type ModalVariants = 'primary' | 'secondary';
+
+export type EnhancedModalLayoutProps = React.PropsWithChildren & {
+  size?: ModalSize;
+  padding?: ModalPadding;
+  className?: string;
+  hotkeyScope?: ModalHotkeyScope;
+  onEnter?: () => void;
+  modalVariant?: ModalVariants;
+} & (
+    | { isClosable: true; onClose: () => void }
+    | { isClosable?: false; onClose?: never } // or the absence of `isClosable`
+  );
+
+const modalAnimation = {
+  hidden: { opacity: 0 },
+  visible: { opacity: 1 },
+  exit: { opacity: 0 },
+};
+
+// This component should be used over Modal when seeking a modal feel without modal state (hotkeyScope etc)
+export const EnhancedModalLayout = ({
```

Comment:
I'll review the PR more in-depth to better understand what you did, but at first look, it isn't very clear what the intention is here. You named the component "EnhancedModalLayout" which is quite generic and not descriptive enough (proof if you were forced to add a comment, if you have a clean API for your component and a clean name, then its purpose should be obvious)

Reply:
Oops, that comment's a mistake.

The point of introducing the new `EnhancedModalLayout` is to get rid of having two different components—one for the modal `ModalLayout` and another for hotscoped `Modal`. Once this `EnhancedModalLayout` is added, we won’t need the existing `ModalLayout` or `Modal`.

After that, we can rename it to either Modal or ModalLayout, whichever fits better. I just called it EnhancedModalLayout for now to keep things clear.